templates/server/web/app_test.go

```diff
@@ -0,0 +1,89 @@
+package web
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"project-name/model"
+	"testing"
+)
+
+var expectedTechnologies = []*model.Technology{
+	{
+		Name:    "Tech1",
+		Details: "Details1",
+	},
+	{
+		Name:    "Tech2",
+		Details: "Details2",
+	},
+}
+
+type MockDb struct {
+	called     int
+	returnTech []*model.Technology
+	returnErr  error
+}
+
+func (m *MockDb) GetTechnologies() ([]*model.Technology, error) {
+	m.called++
+	return m.returnTech, m.returnErr
+}
+
+func TestApp_GetTechnologies(t *testing.T) {
+	mockDb := &MockDb{
+		returnTech: expectedTechnologies,
+	}
+	app := App{d: mockDb}
+
+	req := buildTechnologiesRequest(t)
+	recorder := httptest.NewRecorder()
+
+	app.GetTechnologies(recorder, req)
+
+	verifyDbWasCalledOnce(t, mockDb)
+	status := recorder.Code
+	if status != http.StatusOK {
+		t.Errorf("handler returned wrong status code: got %v want %v", status, http.StatusOK)
+	}
+
+	expected := `[{"name":"Tech1","details":"Details1"},{"name":"Tech2","details":"Details2"}]` + "\n"
+	body := recorder.Body.String()
+	if body != expected {
+		t.Errorf("handler returned unexpected body: got %v want %v", body, expected)
+	}
+}
+
+func TestApp_GetTechnologies_WithDBError(t *testing.T) {
+	mockDb := &MockDb{
+		returnTech: nil,
+		returnErr:  errors.New("unknown error"),
+	}
+	app := App{d: mockDb}
+
+	req := buildTechnologiesRequest(t)
+	recorder := httptest.NewRecorder()
+
+	app.GetTechnologies(recorder, req)
+
+	verifyDbWasCalledOnce(t, mockDb)
+	status := recorder.Code
```

Comment:
II think you can omit extracting Code to a variable. If you rename `recorder` to `w` (see my other comment) it will be short enough to use directly.

```go
if w.Code != http.StatusOK { ... }
```

Reply:
I changed it in both methods